src/zenml/cli/secret.py

```diff
@@ -840,9 +841,8 @@ def create_secret(
         interactive: Whether to use interactive mode to enter the secret values.
         args: The arguments to pass to the secret.
     """
-    name, parsed_args = parse_name_and_extra_arguments(  # type: ignore[assignment]
-        list(args) + [name], expand_args=True
-    )
+    name, parsed_args = parse_secret_name_and_arguements(  # type: ignore[assignment]
```

Comment:
May I suggest a slightly different approach with a somewhat better UX?

Instead of expecting a `--values` or `--values-file` argument here, you could do the following:
* add a single `--values` string click option to the CLI command (just like `--scope` or `--interactive`)
* keep this line unchanged, because now the `values` option will be parsed independently of arguments by click
* parse the `values` option value using the same logic we do with regular arguments: the value could be an inline JSON/YAML encoded string, or it could be a filepath starting with the special character `@`

For example:

```
zenml secret create my-secret --values='{'location': 'Nevada', 'aliens': 'many'}'

zenml secret create my-secret --values=@path/to/my/file.yaml
```

Reply:
You can use this utility to expand the `values` option value into the file contents, if it points to a file: https://github.com/zenml-io/zenml/blob/develop/src/zenml/cli/utils.py#L533

Reply:
agreed, but I got this format structure based on @alex-zenml 's description in below issue:
https://github.com/zenml-io/zenml/issues/620

Reply:
I see what you mean, but that issue is not set in stone, more like a set of guidelines. You can improvise when it makes sense.

Reply:
I think we should also stop user from specifying both at the same time in the command like below - 
`zenml secret create foo --values=JSON --"a"="b"`